components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ResendCodeApiServiceImpl.java

```diff
@@ -64,7 +64,7 @@ public Response resendCodePost(ResendCodeRequestDTO resendCodeRequestDTO) {
         }
 
         if (notificationResponseBean == null || StringUtils.isBlank(notificationResponseBean.getKey())) {
-            return Response.status(Response.Status.CREATED).build();
+            return Response.status(Response.Status.NOT_IMPLEMENTED).build();
```

Comment:
Don't this has to be an Internal Server Error? If the response bean is returning null, it's an error on the server-side right?

Reply:
No neccessarily, because the response stays null if the resend request came for a scenario not checked in this logic, https://github.com/wso2-extensions/identity-governance/blob/b10bdfecdb94975d1931df0837a449d0232e62b3/components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ResendCodeApiServiceImpl.java#L105. Internal server errors in this flow is handled separately then and there.